__tests__/app.test.js

```diff
@@ -0,0 +1,86 @@
+const request = require("supertest")
+const app = require("../db/app")
+const db = require("../db/connection")
+const seed = require("../db/seeds/seed")
+const data = require("../db/data/test-data")
+
+beforeEach(() => {
+    return seed(data)
+})
+
+afterAll(() => {
+    return db.end()
+})
+
+
+describe('GET /api', () => {
+    test('Serves up a JSON representation of all the available endpoints of the api', () => {
+      request(app)
+        .get('/api')
+        .expect(200)
+        .then((response) => {
+            const { body } = response;
+            const { endpoints } = body
+          expect(endpoints).toBeInstanceOf(Object)
+          expect(Object.keys(endpoints)).toHaveLength(3)
+
+          expect(endpoints['GET /api']).toMatchObject({
+            description: expect.any(String),
+          });
+  
+          expect(endpoints['GET /api/topics']).toMatchObject({
+            description: expect.any(String),
+            queries: expect.any(Array),
+            exampleResponse: expect.any(Object),
+          });
+  
+          expect(endpoints['GET /api/articles']).toMatchObject({
+            description: expect.any(String),
+            queries: expect.any(Array),
+            exampleResponse: expect.any(Object),
+          });
+        })
```

Comment:
This test works for now but what will happen when you start adding endpoints to the file 🤔 Do you think you could refactor this to make is dynamic so that it will still pass when you add more? 🚀

Reply:
Hi, It came to mind but I'm still looking for a way to include them all (when the keys/length in each endpoint are not the same)